third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -1175,6 +1177,160 @@ class ScaledBlockedToScaledMFMAF8F6F4 final
   }
 };
 
+class ScaledBlockedToScaledWMMAF8F6F4 final
+    : public OpRewritePattern<triton::DotScaledOp> {
+  int wmmaVersion;
+
+public:
+  ScaledBlockedToScaledWMMAF8F6F4(MLIRContext *context, int wmmaVersion,
+                                  int nonKDim, PatternBenefit benefit = 1)
+      : OpRewritePattern(context, benefit), wmmaVersion(wmmaVersion) {}
+
+  LogicalResult matchAndRewrite(triton::DotScaledOp dotOp,
+                                PatternRewriter &rewriter) const override {
+    int nonKDim = 16;
+    using TensorValue = TypedValue<RankedTensorType>;
+
+    if (wmmaVersion != 3) {
+      return rewriter.notifyMatchFailure(
+          dotOp, "F8F6F4 scaled dot is only natively supported on gfx1251");
+    }
+
+    RankedTensorType oldRetType = dotOp.getType();
+    if (!isa_and_nonnull<BlockedEncodingAttr>(oldRetType.getEncoding())) {
+      return rewriter.notifyMatchFailure(
+          dotOp, "expected blocked encoding result tensor");
+    }
+
+    unsigned rank = oldRetType.getRank();
+    if (rank == 3)
+      return rewriter.notifyMatchFailure(dotOp, "NYI: 3d case");
+
+    TensorValue a = dotOp.getA();
+    TensorValue b = dotOp.getB();
+    TensorValue aScale = dotOp.getAScale();
+    TensorValue bScale = dotOp.getBScale();
+    auto oldShape = oldRetType.getShape();
+
+    ScaleDotElemType aElemType = dotOp.getAElemType();
+    ScaleDotElemType bElemType = dotOp.getBElemType();
+    // TODO: Add more supported types
+    auto supportsTypes = [](ScaleDotElemType elemType) {
+      return elemType == ScaleDotElemType::E2M1;
+    };
+
+    if (!supportsTypes(aElemType) || !supportsTypes(bElemType)) {
+      return rewriter.notifyMatchFailure(dotOp, "Not supported yet mxfp type");
+    }
+
+    MLIRContext *ctx = dotOp.getContext();
+
+    ttg::CTALayoutAttr ctaLayout = ttg::getCTALayout(oldRetType.getEncoding());
+    unsigned numWarps = ttg::lookupNumWarps(dotOp);
+
+    unsigned mDim = 16;
+    unsigned nDim = 16;
+    unsigned kDim = 128;
+
+    auto warpsPerTile = warpsPerTileWMMA(dotOp, oldShape, numWarps, {16, 16});
+
+    auto wmmaEnc = ttg::AMDWmmaEncodingAttr::get(
+        ctx, /*versionMajor=*/wmmaVersion, true, warpsPerTile, ctaLayout,
```

Comment:
Drop `/*versionMajor=*/` as it's outdated--we don't have split major/minor version anymore.